### ISSUE TITLE
Consider adding AppName, AppVersion in the dimensions

### ISSUE THREAD

**Issue by starmessage** [matomo-org/matomo#16027]
I am adapting [SoftMeter](https://www.starmessagesoftware.com/softmeter) to send hits from desktop applications to Matomo. With it, desktop software developers can track the usage of their software application as they do with their website. Very convenient to stay on the same reporting platform.

Matomo doesn't seem to have a dimension for the Application Name and Application Version that is being tracked. 
https://developer.matomo.org/api-reference/tracking-api
In GA there are the AppName and AppVersion dimensions.

An alternative would be to use the "Browser" dimension and populate it with the AppName + Version. But looking at the Matomo repo, it seems that the browser is a closed list of "known" browsers so it cannot accept new values on the fly.

Another thought would be to send the AppName and AppVersion in custom dimensions, but I guess this would require every developer to do this configuration change also in his Matomo installation. This is a deal-breaker as I want the developers to immediately see the usage reports in an out-of-the-box installation of Matomo.

So, my suggestion is to add AppName and AppVersion in the standard dimensions.
Optionally, also add AppLicense, AppEdition.
Optionally, make the Browser name field, a free text field that is parsed either by the UserAgent or have a specific parameter to send it to Matomo.

Love to hear your feedback and any suggestions if there is another way of dealing with the AppName and AppVersion in the current Matomo version.

**Comment by tsteur** [comment]
Hi @starmessage thanks for creating this issue. I reckon in this case you could use Custom 
Dimensions for this purpose? See https://plugins.matomo.org/CustomDimensions

**Comment by starmessage** [comment]
Yes, custom dimensions can be one solution but it requires extra effort to setup.

One of the key features of SoftMeter is that it can be implemented on the same day. So I want to take out any unnecessary steps. Like configuring Matomo which will be a new system for them. 

I would suggest that the Matomo team gives another thought to my suggestion to include these fields (+ the needed reports on the left navigation) because it is a great opportunity to expand Matomo to cover also the application analytics domain (not only Website statistics).
Google Analytics did this expansion and they have two kinds of "Reporting views", the Website and the Mobile App.
Here is an old list I made about application analytics tools for desktop and mobile software applications: 
https://www.starmessagesoftware.com/blog/providers-runtime-analytics-usage-statistics-windows-macosx
Matomo has everything in place and just misses a few convenience tweaks like the AppName and AppVersion. So it would be a pity to miss this opportunity.
If this gets approved, I can provide more info on the needed functionality.

Thanks !

**Comment by sgiehl** [comment]
@starmessage would #5413 solve your needs?

**Comment by starmessage** [comment]
Dear @sgiehl, #5413 is still open for some years, and it also talks about detection done by Matomo. 
Instead, the appName and appVersion should be sent by the application being tracked to the Matomo server via the tracking API. (SoftMeter will do that)
AppName should be free text (e.g 30 chars) and not depend on the code of Matomo in order to be defined. 
To save DB space, it could be modelled as a long int with a lookup to the exact AppName. Clean up can be done periodically removing AppNames with low visits and assign them to an AppName "Other" by a fixed appNameID (e.g. 0).
I am tempted to go to technical details, but I think this is mainly a marketing decision for the Matomo team to take.

![ga-application-version-usage-statistics_0](https://user-images.githubusercontent.com/4344442/84020982-70ba6200-a98c-11ea-92ba-1f4db44538ec.jpg)
As a use case example (to explain better the need), this screenshot is from a report using the appVersion to inform the developers if their customers upgrade the software to the new releases or stay with the old installed versions. (image taken from [here](https://www.starmessagesoftware.com/faq-page/what-is-application-analytics-windows-macos-ios-software))

**Comment by tsteur** [comment]
BTW when creating a new meaurable of type Mobile App an AppID can be already added: 
![image](https://user-images.githubusercontent.com/273120/84076233-8467d600-aa29-11ea-9a7e-e4d8363a0723.png)

The name field would basically represent the app name.

Not sure the app name would be a dimension or why that would be useful unless you were to track multiple apps into one app in Matomo?

**Comment by starmessage** [comment]
@tsteur, indeed the recommendation is to use one trackingID per applications. But there are cases where it is better to track multiple applications on one TrasckingID. e.g.
- you have a family of related or complementary software and you want to see the timelines of their users in the same graph
- your marketing promotes all your software products so you want to see the total users, total installations, total uninstalls in one report.
- segmenting and focusing on one appName is easy, so you watch only one Matomo property instead of switching between them to see if something (e.g. unexpected increase, decrease in users) has happened.

**Comment by tsteur** [comment]
Just btw for this we'd usually recommend tracking each app separately and then using Roll-Up Reporting.

**Comment by starmessage** [comment]
Thanks, I am keeping notes of your hints and will assemble them in a step by step guide for the SoftMeter customers to follow the recommendations when configuring Matomo.
